tests/test_helper.tcl

```diff
@@ -857,9 +857,24 @@ proc read_from_replication_stream {s} {
 }
 
 proc assert_replication_stream {s patterns} {
+    set errors 0
+    set values {}
     for {set j 0} {$j < [llength $patterns]} {incr j} {
-        assert_match [lindex $patterns $j] [read_from_replication_stream $s]
+        set pattern [lindex $patterns $j]
+        set value [read_from_replication_stream $s]
+        lappend values $value
+        if {![string match $pattern $value]} { incr errors }
     }
+
+    if {$errors == 0} { return }
+
+    puts "replication stream start:"
+    foreach value $values { puts "$value" }
+    puts "replication stream end."
```

Comment:
i'd rather avoid `puts` here, and send all the details to the assertion (to be printed by the main tcl process rather than getting mixed with other tests.

Reply:
i think that's ok.. maybe we can fix the context (frame) though?

Reply:
now, it become this (can't think of any more beautiful print...):
```
Testing unit/expire
[err]: GETEX propagate as to replica as PERSIST, DEL, or nothing in tests/unit/expire.tcl
Expected '{select 9} {set foo bar PXAT 1678292121527} {persist foo} {del foo}' to match '{select *} {set foo bar PXAT *} {persist foo} {del foo1}' type eval line 10 cmd {assert_replication_stream $repl {
               {select *}
               {set foo bar PXAT *}
               {persist foo}
               {del foo1}
            }} proc ::test
[1/1 done]: unit/expire (0 seconds)

                   The End

Execution time of different units:
  0 seconds - unit/expire

!!! WARNING The following tests failed:

*** [err]: GETEX propagate as to replica as PERSIST, DEL, or nothing in tests/unit/expire.tcl
Expected '{select 9} {set foo bar PXAT 1678292121527} {persist foo} {del foo}' to match '{select *} {set foo bar PXAT *} {persist foo} {del foo1}' type eval line 10 cmd {assert_replication_stream $repl {
               {select *}
               {set foo bar PXAT *}
               {persist foo}
               {del foo1}
            }} proc ::test
Cleanup: may take some time... OK
```